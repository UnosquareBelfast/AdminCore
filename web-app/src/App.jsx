import React from 'react';
import Header from './components/Header';
import withAuth from './hoc/withAuth';
import { Dashboard } from './pages';

class App extends React.Component {
  render() {
    return (
      <div>
        <Header user={this.props.user} history={this.props.history} />
        <Dashboard user={this.props.user} />
      </div>
    );
  }
}

export default withAuth(App);
