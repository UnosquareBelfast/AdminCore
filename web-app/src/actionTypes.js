//USER
export const UPDATE_USER = 'UPDATE_USER';
export const RESET_USER = 'RESET_USER';

// DASHBOARD

export const SELECT_EVENT = 'SELECT_EVENT';
export const UPDATE_EVENT_DURATION = 'UPDATE_EVENT_DURATION';
export const TOGGLE_BOOKING_MODAL = 'TOGGLE_BOOKING_MODAL';
export const SET_IS_BEING_UPDATED = 'SET_IS_BEING_UPDATED';
export const SET_CALENDAR_EVENTS = 'SET_CALENDAR_EVENTS';
export const SET_ERROR = 'SET_ERROR';

// LOADING

export const SET_LOADING = 'SET_LOADING';
