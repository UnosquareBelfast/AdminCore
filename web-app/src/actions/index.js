export {
  fetchEvents,
  filterEventsByEmployeeId,
  updateBookingAndDuration,
  toggleBookingModal,
  updateBooking,
  updateBookingDuration,
} from './dashboard';
