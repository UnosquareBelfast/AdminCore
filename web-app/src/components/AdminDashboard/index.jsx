import React from 'react';
import { Link } from 'react-router-dom';
import { InnerLayout } from './styled';

const AdminDashboard = () => {
  return (
    <InnerLayout>
      <h2>Admin Dashboard</h2>
      <div className="columnWrap">
        <div>
          <h3>Employees</h3>
          <Link to="/admin/employees">View Employees</Link>
        </div>
        <div>
          <h3>Holidays</h3>
          <Link to="/admin/holidays">View All Holidays</Link>
          <Link to="/admin/pendingHolidays">View Pending Holidays</Link>
        </div>
        <div>
          <h3>Clients</h3>
          <Link to="">View All Holidays</Link>
          <Link to="">View Pending Holidays</Link>
        </div>
        <div>
          <h3>Contracts</h3>
          <Link to="">View All Holidays</Link>
          <Link to="">View Pending Holidays</Link>
        </div>
      </div>
    </InnerLayout>
  );
};

export default AdminDashboard;
