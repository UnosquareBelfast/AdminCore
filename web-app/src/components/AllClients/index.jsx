import React, { Fragment } from 'react';
import { PropTypes as PT } from 'prop-types';
import container from './container';
import { Table, Button } from '../common';
import { ActiveDot, CTA } from './styled';
import employeeStatus from '../../utilities/employeeStatus';

export const AllClients = ({
  clients,
  viewClient,
  archive,
  createNewClient,
}) => {
  let clientList;
  if (clients.length > 0) {
    clientList = clients.map(client => (
      <tr key={client.clientId}>
        <td>{client.clientName}</td>
        <td>{client.clientStatusDescription}</td>
        <td>
          <ActiveDot active={client.clientStatusId === employeeStatus.ACTIVE} />
          {client.clientStatusId === employeeStatus.ACTIVE
            ? 'Active'
            : 'Inactive'}
        </td>
        <td>
          <button onClick={() => viewClient(client.clientId)}>
            Edit Client
          </button>
          {client.clientStatusId === employeeStatus.ACTIVE ? (
            <button className="success" onClick={() => archive(client)}>
              Archive
            </button>
          ) : (
            <button className="error" onClick={() => archive(client, true)}>
              Activate
            </button>
          )}
        </td>
      </tr>
    ));
  } else {
    clientList = (
      <tr>
        <td>No Clients</td>
        <td />
        <td />
        <td />
      </tr>
    );
  }
  return (
    <Fragment>
      <h2>All Clients</h2>
      <Table
        tableHeaders={['Name', 'Description', 'Status', ' ']}
        tableRows={clientList}
      />
      <CTA>
        <Button onClick={() => createNewClient()} label="New Client" />
      </CTA>
    </Fragment>
  );
};

AllClients.propTypes = {
  createNewClient: PT.func,
  clients: PT.array,
  archive: PT.func.isRequired,
  viewClient: PT.func.isRequired,
};

export default container(AllClients);
