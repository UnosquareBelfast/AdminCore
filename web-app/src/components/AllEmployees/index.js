import React, { Fragment } from 'react';
import { PropTypes as PT } from 'prop-types';
import { UserList, UserModal } from '../';
import { Button } from '../common';
import { CornerButton } from '../common_styled';
import container from './container';

const AllEmployees = ({ history, users, selectUser, selectedUser }) => {
  return (
    <Fragment>
      <CornerButton>
        <Button
          onClick={() => history.replace('/admin/employees/new')}
          label="New Employee"
        />
      </CornerButton>
      <UserModal
        user={selectedUser}
        closeModal={() => selectUser({})}
        history={history}
      />
      <h2>Employees</h2>
      <UserList
        history={history}
        users={users}
        columns={['fullName', 'email', 'role', 'location']}
        onRowClick={employee => selectUser(employee)}
      />
    </Fragment>
  );
};

AllEmployees.propTypes = {
  history: PT.object.isRequired,
  users: PT.array.isRequired,
  selectUser: PT.func.isRequired,
  selectedUser: PT.object.isRequired,
};

export default container(AllEmployees);
