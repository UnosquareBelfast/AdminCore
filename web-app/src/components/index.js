export { default as AdminDashboard } from './AdminDashboard';
export { default as AllClients } from './AllClients';
export { default as AllHolidays } from './AllHolidays';
export { default as BigCalendarToolbar } from './BigCalendarToolbar';
export { default as BookedLeave } from './BookedLeave';
export { default as BookingCalendar } from './BookingCalendar';
export { default as BookingModal } from './BookingModal';
export { default as CreateClient } from './CreateClient';
export { default as CreateContract } from './CreateContract';
export { default as ClientModal } from './ClientModal';
export { default as CreateTeam } from './CreateTeam';
export { default as CreateUser } from './CreateUser';
export { default as Legend } from './Legend';
export { default as PendingHolidays } from './PendingHolidays';
export { default as RangeDayPicker } from './common/RangeDayPicker';
export { default as RequestHoliday } from './RequestHoliday';
export { default as TakenLeave } from './TakenLeave';
export { default as TeamView } from './TeamView';
export { default as UserDetails } from './UserDetails';
export { default as UserModal } from './UserModal';
export { default as ViewContracts } from './ViewContracts';
export { default as ViewTeams } from './ViewTeams';
export { default as HolidayModal } from './HolidayModal';
export { default as AllEmployees } from './AllEmployees';
export { default as DataTable } from './DataTable';
