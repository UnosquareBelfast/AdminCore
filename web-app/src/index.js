import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import App from './App';
import { Login } from './pages';

ReactDOM.render(
  <Router>
    <Switch>
      <Route exact path="/login" component={Login} />
      <Route path="/" component={App} />
    </Switch>
  </Router>,
  document.getElementById('app')
);
