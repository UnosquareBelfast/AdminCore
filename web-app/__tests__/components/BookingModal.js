import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import { BookingModal } from '../../src/components';

const booking = {
  title: 'Fake Booking',
  holidayStatusDescription: 'Pending',
  start: moment('2018-01-01 09:00:00'),
  end: moment('2018-01-01 09:00:00'),
  isHalfday: false,
  isWFH: false,
  duration: 1,
};

xdescribe('BookingModal', () => {
  it('renders correctly', () => {
    const mockOnClick = jest.fn();
    const mockChangeStart = jest.fn();
    const mockChangeEnd = jest.fn();
    const mockChangeHalfDay = jest.fn();
    const mockChangeWFH = jest.fn();
    const mockUpdateHoliday = jest.fn();
    const mockCancelHoliday = jest.fn();

    const wrapper = shallow(
      <BookingModal
        closeModal={mockOnClick}
        showModal
        booking={booking}
        changeStart={mockChangeStart}
        changeEnd={mockChangeEnd}
        changeHalfday={mockChangeHalfDay}
        requestHoliday={mockOnClick}
        cancelHoliday={mockCancelHoliday}
        updateHoliday={mockUpdateHoliday}
        changeWFH={mockChangeWFH}
      />,
    );

    expect(wrapper.exists());
    expect(wrapper).toMatchSnapshot();
  });

  it('closeModal calls closeModal prop', () => {
    const mockOnClick = jest.fn();

    const wrapper = shallow(
      <BookingModal
        closeModal={mockOnClick}
        showModal
        booking={{}}
        changeStart={() => {}}
        changeEnd={() => {}}
        changeHalfday={() => {}}
        requestHoliday={() => {}}
        cancelHoliday={() => {}}
        updateHoliday={() => {}}
        changeWFH={() => {}}
      />,
    );

    expect(mockOnClick.mock.calls.length).toEqual(0);
    wrapper.find('#closeBookingModal').simulate('click');
    expect(mockOnClick.mock.calls.length).toEqual(1);

    expect(wrapper.exists());
  });

  describe('getTotalDays', () => {
    it('getTotalDays returns the correct amount of days leave when duration is 1 and halfDay is false', () => {
      const wrapper = shallow(
        <BookingModal
          closeModal={() => {}}
          showModal
          booking={{ ...booking, duration: 1 }}
          changeStart={() => {}}
          changeEnd={() => {}}
          changeHalfday={() => {}}
          requestHoliday={() => {}}
          cancelHoliday={() => {}}
          updateHoliday={() => {}}
          changeWFH={() => {}}
        />,
      );

      expect(wrapper.exists());
      expect(wrapper.find('#totalDaysToBook').text()).toEqual('Total days: 1');
    });
    it('getTotalDays returns the correct amount of days leave when duration is 2 and halfDay is false', () => {
      const wrapper = shallow(
        <BookingModal
          closeModal={() => {}}
          showModal
          booking={{ ...booking, duration: 2 }}
          changeStart={() => {}}
          changeEnd={() => {}}
          changeHalfday={() => {}}
          requestHoliday={() => {}}
          cancelHoliday={() => {}}
          updateHoliday={() => {}}
          changeWFH={() => {}}
        />,
      );

      expect(wrapper.exists());
      expect(wrapper.find('#totalDaysToBook').text()).toEqual('Total days: 2');
    });
    it('getTotalDays returns the correct amount of days leave when duration is 3 and halfDay is false', () => {
      const wrapper = shallow(
        <BookingModal
          closeModal={() => {}}
          showModal
          booking={{ ...booking, duration: 3 }}
          changeStart={() => {}}
          changeEnd={() => {}}
          changeHalfday={() => {}}
          requestHoliday={() => {}}
          cancelHoliday={() => {}}
          updateHoliday={() => {}}
          changeWFH={() => {}}
        />,
      );

      expect(wrapper.exists());
      expect(wrapper.find('#totalDaysToBook').text()).toEqual('Total days: 3');
    });
    it('getTotalDays returns the correct amount of days leave when duration is 3 and halfDay is true', () => {
      const wrapper = shallow(
        <BookingModal
          closeModal={() => {}}
          showModal
          booking={{ ...booking, duration: 1, isHalfday: true }}
          changeStart={() => {}}
          changeEnd={() => {}}
          changeHalfday={() => {}}
          requestHoliday={() => {}}
          cancelHoliday={() => {}}
          updateHoliday={() => {}}
          changeWFH={() => {}}
        />,
      );

      expect(wrapper.exists());
      expect(wrapper.find('#totalDaysToBook').text()).toEqual(
        'Total days: 0.5',
      );
    });
    it('getTotalDays returns 0 when isWFH is true', () => {
      const wrapper = shallow(
        <BookingModal
          closeModal={() => {}}
          showModal
          booking={{ ...booking, duration: 1, isWFH: true }}
          changeStart={() => {}}
          changeEnd={() => {}}
          changeHalfday={() => {}}
          requestHoliday={() => {}}
          cancelHoliday={() => {}}
          updateHoliday={() => {}}
          changeWFH={() => {}}
        />,
      );

      expect(wrapper.exists());
      expect(wrapper.find('#totalDaysToBook').text()).toEqual('Total days: 0');
    });
  });

  describe('getActions', () => {
    it('returns a Button with the text "Add" when isWHF is true booking.title is undefined', () => {
      const mockRequestHoliday = jest.fn();

      const wrapper = shallow(
        <BookingModal
          closeModal={() => {}}
          showModal
          booking={{ ...booking, title: undefined, isWFH: true }}
          changeStart={() => {}}
          changeEnd={() => {}}
          changeHalfday={() => {}}
          requestHoliday={mockRequestHoliday}
          cancelHoliday={() => {}}
          updateHoliday={() => {}}
          changeWFH={() => {}}
        />,
      );

      expect(wrapper.find('#requestHolidayBtn').exists()).toBe(true);
      expect(wrapper.find('#requestHolidayBtn').prop('onClick')).toEqual(
        mockRequestHoliday,
      );
      expect(wrapper.find('#requestHolidayBtn').prop('label')).toEqual('Add');
    });
    it('returns a Button with the text "Request" when isWHF is false booking.title is undefined', () => {
      const mockRequestHoliday = jest.fn();

      const wrapper = shallow(
        <BookingModal
          closeModal={() => {}}
          showModal
          booking={{ ...booking, title: undefined, isWFH: false }}
          changeStart={() => {}}
          changeEnd={() => {}}
          changeHalfday={() => {}}
          requestHoliday={mockRequestHoliday}
          cancelHoliday={() => {}}
          updateHoliday={() => {}}
          changeWFH={() => {}}
        />,
      );

      expect(wrapper.find('#requestHolidayBtn').exists()).toBe(true);
      expect(wrapper.find('#requestHolidayBtn').prop('onClick')).toEqual(
        mockRequestHoliday,
      );
      expect(wrapper.find('#requestHolidayBtn').prop('label')).toEqual(
        'Request',
      );
    });
  });
});
