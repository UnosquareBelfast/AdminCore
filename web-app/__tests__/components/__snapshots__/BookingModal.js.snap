// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingModal renders correctly 1`] = `
<Modal>
  <styled.div>
    <span
      id="closeBookingModal"
      onClick={[MockFunction]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              352,
              512,
              Array [],
              "f00d",
              "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
            ],
            "iconName": "times",
            "prefix": "fas",
          }
        }
        listItem={false}
        mask={null}
        name=""
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
       Close
    </span>
    <h1>
      Booking
    </h1>
    <styled.div>
      <h4>
        Fake Booking
      </h4>
      <span>
        <styled.div />
      </span>
    </styled.div>
    <styled.div>
      <div>
        <label>
          Starting:
        </label>
        <DatePicker
          allowSameDay={false}
          dateFormat="L"
          dateFormatCalendar="MMMM YYYY"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          monthsShown={1}
          onBlur={[Function]}
          onChange={[MockFunction]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          preventOpenOnFocus={false}
          selected={"2018-01-01T09:00:00.000Z"}
          shouldCloseOnSelect={true}
          showTimeSelect={false}
          timeCaption="Time"
          timeIntervals={30}
          withPortal={false}
        />
      </div>
      <div>
        <label>
          Ending:
        </label>
        <DatePicker
          allowSameDay={false}
          dateFormat="L"
          dateFormatCalendar="MMMM YYYY"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          monthsShown={1}
          onBlur={[Function]}
          onChange={[MockFunction]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          preventOpenOnFocus={false}
          selected={"2018-01-01T09:00:00.000Z"}
          shouldCloseOnSelect={true}
          showTimeSelect={false}
          timeCaption="Time"
          timeIntervals={30}
          withPortal={false}
        />
      </div>
    </styled.div>
    <form>
      <label>
        <input
          checked={false}
          disabled={false}
          onChange={[MockFunction]}
          type="checkbox"
        />
        Half-Day
      </label>
      <label>
        <input
          checked={false}
          disabled={false}
          onChange={[MockFunction]}
          type="checkbox"
        />
        WFH
      </label>
    </form>
    <p
      id="totalDaysToBook"
    >
      Total days: 
      1
    </p>
    <styled.div>
      <Button
        id="updateHolidayBtn"
        label="Update"
        onClick={[MockFunction]}
      />
      <Button
        id="cancelHolidayBtn"
        label="Cancel Booking"
        onClick={[MockFunction]}
      />
    </styled.div>
  </styled.div>
</Modal>
`;
